causalml/feature_selection/filters.py

```diff
@@ -39,26 +42,64 @@ def _filter_F_one_feature(data, treatment_indicator, feature_name, y_name):
             [treatment_indicator, feature_name]
         ].product(axis=1)
 
+        if order not in [1, 2, 3]:
+            raise Exception("ValueError: order argument only takes value 1,2,3.")
+
+        if order == 1:
+            pass
+        elif order == 2:
+            x_tmp_name = "{}_o{}".format(feature_name, order)
+            X[x_tmp_name] = X[[feature_name]] ** order
+            X["{}-{}".format(treatment_indicator, x_tmp_name)] = X[
+                [treatment_indicator, x_tmp_name]
+            ].product(axis=1)
+        elif order == 3:
+            x_tmp_name = "{}_o{}".format(feature_name, 2)
+            X[x_tmp_name] = X[[feature_name]] ** 2
+            X["{}-{}".format(treatment_indicator, x_tmp_name)] = X[
+                [treatment_indicator, x_tmp_name]
+            ].product(axis=1)
+
+            x_tmp_name = "{}_o{}".format(feature_name, order)
+            X[x_tmp_name] = X[[feature_name]] ** order
+            X["{}-{}".format(treatment_indicator, x_tmp_name)] = X[
+                [treatment_indicator, x_tmp_name]
+            ].product(axis=1)
+
         model = sm.OLS(Y, X)
         result = model.fit()
 
-        F_test = result.f_test(np.array([0, 0, 0, 1]))
+        if order == 1:
+            F_test = result.f_test(np.array([0, 0, 0, 1]))
+        elif order == 2:
+            F_test = result.f_test(np.array([[0, 0, 0, 1, 0, 0], [0, 0, 0, 0, 0, 1]]))
+        elif order == 3:
+            F_test = result.f_test(
```

Comment:
@zhenyuz0500 can you remind me why is the _r_matrix_ configured in this way? Maybe we can add comment here too

Reply:
The r_matrix assumes the linear combination of each row equals 0, and the tests of columns are done jointly (as one test).
For example, when order=2, then the linear model will be: a + b1 * I_treatment + b2 * x + b3 * x * I_treatment + b4 * x^2 + b5 * x^2 * I_treatment
we want to test H0: b3==0 and b5 ==0 vs H1: b3!=0 or b5 !=0
then it translates to test H0: [0, 0, 0, 1, 0, 0] * [a, b1, b2, b3, b4,b5]' = 0 and [0, 0, 0, 0, 0, 1] * [a, b1, b2, b3, b4, b5]' = 0

reference: https://www.statsmodels.org/dev/generated/statsmodels.regression.linear_model.RegressionResults.f_test.html

Reply:
I see, thanks!

Reply:
thanks for the review!